src/Compilers/Core/Portable/MetadataReader/MetadataTypeName.cs

```diff
@@ -285,6 +360,19 @@ public ImmutableArray<string> NamespaceSegments
             }
         }
 
+        public ImmutableArray<ReadOnlyMemory<char>> NamespaceSegmentsMemory
+        {
+            get
+            {
+                if (_namespaceSegmentsMemory.IsDefault)
+                {
+                    _namespaceSegmentsMemory = MetadataHelpers.SplitQualifiedName(NamespaceNameMemory);
```

Comment:
>MetadataHelpers.SplitQualifiedName(NamespaceNameMemory) [](http://example.com/codeflow?start=47&length=55)

Similar comment about avoiding doing the work.

Reply:
Yup.  Done.